components/SetUpStaking/StakingStepper.jsx

```diff
@@ -0,0 +1,165 @@
+import { useState, useEffect } from 'react'
+import { useSelector } from 'react-redux';
+import { Flex, Steps, Typography, Button, Skeleton } from 'antd';
+import Link from 'next/link';
+import { notifyError } from '@autonolas/frontend-library';
+import { truncateAddress, formatToEth, getAddressFromBytes32 } from 'common-util/functions';
+import ConnectTwitterModal from '../ConnectTwitter/Modal';
+import { useTotalBond, checkAndApproveToken, createAndStake } from './requests';
+
+const { Paragraph, Text } = Typography;
+
+const XDAI_FOR_GAS = 2;
+
+const ConnectTwitter = ({ account }) => {
+  if (account) {
+    return (
+      <Text type="secondary">
+        Your connected Twitter account:{' '}
+        <a href={`https://twitter.com/${account}`} target="_blank">@{account} ↗</a>
+      </Text>
+    )
+  }
+  return (
+    <>
+      <Paragraph type="secondary">
+        Link your Twitter account to your Contribute profile.
+      </Paragraph>
+      <ConnectTwitterModal/>
+    </>
+  )
+}
+
+const SetUpAndStake = ({ disabled, twitterId, onNextStep }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [multisig, setMultisig] = useState(null);
+
+  const account = useSelector((state) => state?.setup?.account);
+
+  const { totalBond, isLoading: isTotalBondLoading } = useTotalBond()
+
+  const handleSetUpAndStake = async () => {
+    if (!account) return;
+    if (!twitterId) return;
+    if (!totalBond) return;
+
+    setIsLoading(true);
+
+    try {
+      await checkAndApproveToken({
+        account,
+        amountToApprove: totalBond,
+      })
+      
+      const result = await createAndStake({
+        socialId: twitterId,
+      })
+
+      if (result) {
+        const logs = result.logs;
+        const createdAndStakedEvent = logs[logs.length - 1];
+        const multisig = getAddressFromBytes32(createdAndStakedEvent.topics[3]);
+        // TODO: write multisig to ceramic
+        setMultisig(multisig)
+        onNextStep()
+      }
+
+    } catch (error) {
+      notifyError('Error: could not set up & stake');
+      console.error(error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  if (multisig) {
+    return (
+      <Text type="secondary">
+        Your contribute service Safe address:{' '}
+        <a href={`https://gnosisscan.io/address/${multisig}`} target="_blank">
```

Comment:
does this need to be base explorer? read further down, and looks like we're using Base? 😕